File: FritzBox/Model/Device.cs
using System.Xml.Serialization;
using Newtonsoft.Json;

namespace RaspPiTest.FritzBox.Model
{
    [JsonObject(MemberSerialization.OptIn)]
    public class Device
    {
        [XmlElement("present")]
        public byte IsPresent { get; set; }

        [XmlElement("name"), JsonProperty("name")]
        public string Name { get; set; }

        [XmlElement("temperature"), JsonProperty("deviceTemperature")]
        public DeviceTemperature Temperature { get; set; }

        [XmlElement("hkr"), JsonProperty("thermostat")]
        public Thermostat Thermostat { get; set; }

        [XmlAttribute("identifier"), JsonProperty("ain")]
        public string Identifier { get; set; }

        [XmlAttribute("id")]
        public byte Id { get; set; }

        [XmlAttribute("functionbitmask")]
        public ushort Functions { get; set; }

        [XmlAttribute("fwversion")]
        public decimal FwVersion { get; set; }

        [XmlAttribute("manufacturer"), JsonProperty("manufacturer")]
        public string Manufacturer { get; set; }

        [XmlAttribute("productname"), JsonProperty("productName")]
        public string ProductName { get; set; }
    }
}

File: App/pidrei-ng-app/src/shared/classes/device.ts
import { Thermostat } from 'src/shared/classes/thermostat';
import { DeviceTemperature } from 'src/shared/classes/device-temperature';

export class Device {
    constructor() {
        this.deviceTemperature = new DeviceTemperature();
        this.thermostat = new Thermostat();
    }

    public name: string;
    public ain: string;
    public deviceTemperature: DeviceTemperature;
    public thermostat: Thermostat;
    public manufacturer: string;
    public productName: string;

    public assign(other: Device): Device {
        this.ain = other.ain;
        this.name = other.name;
        this.deviceTemperature = this.deviceTemperature || new DeviceTemperature();
        this.deviceTemperature.temperature = other.deviceTemperature.temperature;
        this.deviceTemperature.offset = other.deviceTemperature.offset;
        this.thermostat.assign(other.thermostat);
        this.manufacturer = other.manufacturer;
        this.productName = other.productName;

        return this;
    }
}
